### PR TITLE
Add count of one time keys to sync stream

### PR DESCRIPTION
**Pull request by erikjohnston**
This adds the following to a sync result:

```json
{
    "device_one_time_keys_count": {
        "signed_curve25519": 50
    },
    ...
}
```

### CODE REVIEW

**Comment by erikjohnston** [comment]
> looks ok, but count_e2e_one_time_keys has a cache, and I'm unconvinced it will be correctly invalidated in the synchotrons when a key is claimed in the main process.

Oops, yeah. It now streams the invalidations to workers.

**Comment by erikjohnston** [comment]
(The sqlite failure is spurious as it built on a box that didn't have tox installed)

**Comment by erikjohnston** [comment]
But i used grep and everything! :(

File: synapse/storage/end_to_end_keys.py [hunk 1]
```diff
@@ -240,6 +240,9 @@ def _claim_e2e_one_time_keys(txn):
                 txn.call_after(
                     self.count_e2e_one_time_keys.invalidate, (user_id, device_id,)
```
Comment (richvdh) [inline synapse/storage/end_to_end_keys.py]: now redundant?

**Comment by richvdh** [comment]
> But i used grep and everything! :(

Apparently you fail at grep.

**Comment by erikjohnston** [comment]
> Apparently you fail at grep.

And editing code too. Sigh. *applies more coffee*

**Comment by bmarty** [comment]
(FTR https://github.com/vector-im/riotX-android/pull/737)

**Pull request merged**